libs/application/templates/inheritance-report/src/fields/ReportFieldsRepeater/index.tsx

```diff
@@ -0,0 +1,286 @@
+  Box,
+  GridColumn,
+  GridRow,
+  Button,
+  Text,
+  Input,
+} from '@island.is/island-ui/core'
+import { Answers } from '../../types'
+import * as styles from '../styles.css'
+import { getValueViaPath } from '@island.is/application/core'
+import { formatCurrency } from '@island.is/application/ui-components'
+import { currencyStringToNumber } from '../../lib/utils/currencyStringToNumber'
+import { Skattleysismörk } from '../../lib/constants'
+import { useLocale } from '@island.is/localization'
+import { m } from '../../lib/messages'
+
+type RepeaterProps = {
+  field: {
+    props: {
+      fields: Array<object>
+      repeaterButtonText: string
+      repeaterHeaderText: string
+      sumField: string
+      fromExternalData?: string
+    }
+  }
+}
+
+export const ReportFieldsRepeater: FC<
+  FieldBaseProps<Answers> & RepeaterProps
+> = ({ application, field }) => {
+  const { id, props } = field
+  const { fields, append, remove } = useFieldArray<any>({
+    name: id,
+  })
+
+  const { setValue } = useFormContext()
+  const { formatMessage } = useLocale()
+  const answersValues = getValueViaPath(
+    application.answers,
+    id,
+  ) as Array<object>
+
+  /* ------ Stocks ------ */
+  const [rateOfExchange, setRateOfExchange] = useState(0)
+  const [faceValue, setFaceValue] = useState(0)
+  const [index, setIndex] = useState('0')
+
+  /* ------ Heirs ------ */
+  const [percentage, setPercentage] = useState(0)
+  const [taxFreeInheritance, setTaxFreeInheritance] = useState(0)
+  const [inheritance, setInheritance] = useState(0)
+  const [taxableInheritance, setTaxableInheritance] = useState(0)
+  const [inheritanceTax, setInheritanceTax] = useState(0)
+
+  /* ------ Total ------ */
+  const answersValuesTotal = answersValues?.length
+    ? answersValues.reduce((a: number, o: any) => {
+        return a + Number(o[props.sumField])
+      }, 0)
+    : 0
+
+  const [valueArray, setValueArray] = useState<Array<number>>(
+    answersValues?.length
+      ? answersValues.map((v: any) => Number(v[props.sumField]))
+      : [],
+  )
+
+  const [total, setTotal] = useState(
+    answersValues?.length ? answersValuesTotal : 0,
+  )
+
+  const handleAddRepeaterFields = () => {
+    const values = props.fields.map((field: object) => {
+      return Object.values(field)[1]
+    })
+
+    const repeaterFields = values.reduce((acc: any, elem: any) => {
+      acc[elem] = ''
+      return acc
+    }, {})
+
+    append(repeaterFields)
+  }
+
+  /* ------ Set total value ------ */
+  useEffect(() => {
+    const addTotal = id.replace('data', 'total')
+    setValue(addTotal, total)
+  }, [id, total, setValue])
+
+  /* ------ Set stocks value ------ */
+  useEffect(() => {
+    setValue(`${index}.value`, String(faceValue * rateOfExchange))
+  }, [faceValue, rateOfExchange, setValue])
+
+  /* ------ Set heirs calculations ------ */
+  useEffect(() => {
+    setTaxFreeInheritance(Skattleysismörk * percentage)
+    setInheritance(15000000 * percentage)
```

Comment:
What is this constant `15000000`?

Reply:
vúps, good spot

Reply:
nooono, i just forgot to replace it with the answers value :D